react/src/views/Transfers/CreateTransferShipment/CreateTransferShipmentView.tsx

```diff
@@ -0,0 +1,174 @@
+  const { triggerError } = useErrorHandling();
+  const { createToast } = useNotification();
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+
+  // variables in URL
+  const baseId = useParams<{ baseId: string }>().baseId!;
+
+  // Query Data for the Form
+  const allFormOptions = useQuery<AllAcceptedTransferAgreementsQuery>(
+    ALL_ORGS_AND_BASES_WITH_TRANSFER_AGREEMENTS_QUERY,
+    {
+      variables: {
+        baseId,
+      },
+    },
+  );
+
+  // Mutation after form submission
+  const [createTransferShipmentMutation, createTransferShipmentMutationState] = useMutation<
+    CreateTransferShipmentMutation,
+    CreateTransferShipmentMutationVariables
+  >(CREATE_SHIPMENT_MUTATION);
+
+  // Prep data for Form
+  const allTransferAgreements = allFormOptions.data?.transferAgreements;
+
+  // eslint-disable-next-line max-len
+  const currentOrganisationLabel = `${allFormOptions?.data?.base?.organisation?.name} - ${allFormOptions?.data?.base?.name}`;
+
+  const partnerOrgsAgreementData = allTransferAgreements
+    ?.filter(
+      (agreement) =>
+        agreement.sourceOrganisation.id === globalPreferences.selectedOrganisationId?.toString() &&
+        (agreement.type === TransferAgreementType.SendingTo ||
+          agreement.type === TransferAgreementType.Bidirectional),
+    )
+    .map(
+      (agreement) =>
+        ({
+          agreementId: agreement.id,
+          specialNote: agreement.comment,
+          orgId: agreement.targetOrganisation.id,
+          orgName: agreement.targetOrganisation.name,
+          orgBases: agreement.targetBases,
+        } as IOrganisationsAgreementsDataData),
+    );
+
+  // Handle Submission
+  const onSubmitCreateShipmentForm = (createTransferShipmentData: ITransferShipmentFormData) => {
+    const agreementId =
+      createTransferShipmentData?.partnerOrganisationSelectedBase.data?.agreementId || null;
+
+    if (agreementId === null) {
+      triggerError({
+        message: "Error while trying to create a new shipment",
+      });
+    } else {
+      createTransferShipmentMutation({
+        variables: {
+          transferAgreementId: parseInt(agreementId, 10),
+          sourceBaseId: parseInt(baseId, 10),
+          targetBaseId: parseInt(createTransferShipmentData.partnerOrganisation.value, 10),
+        },
+      })
+        .then((mutationResult) => {
+          if (mutationResult.errors) {
+            triggerError({
+              message: "Error while trying to create a new shipment",
+            });
+          } else {
+            createToast({
+              title: `Transfer Shipment ${mutationResult.data?.createShipment?.id}`,
+              type: "success",
+              message: "Successfully created a new shipment",
+            });
+
+            navigate(`/bases/${baseId}/transfers/shipments`);
+          }
+        })
+        .catch((err) => {
+          triggerError({
+            message: "Your changes could not be saved!",
```

Comment:
Prob the message should be different, like `Error while trying to create a new shipment`

Reply:
@HaGuesto Sure, I'll update the test plan error message text and change the message.